spacelift/data_aws_integrations.go

```diff
@@ -0,0 +1,109 @@
+package spacelift
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/structs"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/validations"
+)
+
+func dataAWSIntegrations() *schema.Resource {
+	return &schema.Resource{
+		Description: "`spacelift_aws_integration` represents a list of all the AWS integrations in the Spacelift account",
+		ReadContext: dataAWSIntegrationsRead,
+		Schema: map[string]*schema.Schema{
+			"integrations": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"integration_id": {
+							Type:             schema.TypeString,
+							Description:      "Immutable ID of the integration.",
+							Optional:         true,
```

Comment:
Why are the `integration_id` and `name` optional? Shouldn't they just be computed - it doesn't look like we're using the values for querying.

Reply:
Correct, I removed those